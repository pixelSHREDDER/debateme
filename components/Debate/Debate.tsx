'use client'

import { useUser } from '@auth0/nextjs-auth0/client'
import NewTurn from '@/components/Turn/NewTurn'
import { TDebate } from '@/lib/prisma-types'
import Turns from '@/components/Turn/Turns'
import { useMemo } from 'react'
import JoinDebate from '../Invite/JoinDebate'
import Invite from '../Invite/Invite'

interface IDebate {
  debate: TDebate | null,
  updateDebate: Function
}

export default function Debate({ debate, updateDebate }: IDebate) {
  const { user, error, isLoading } = useUser()

  const isCooldownActive = useMemo(() => {
    if (!debate) {
      return false
    }

    const lastTurn = debate.turn?.at(-1)

    if (
      !debate.turn ||
      !debate.turn.length ||
      !lastTurn ||
      lastTurn.userSub === user?.sub
    ) {
      return false
    }

    return (
      Date.now() - new Date(lastTurn.createdAt).getTime() < (debate.cooldownMins * 100_000)
    )
  }, [debate, user?.sub])

  const isItYourTurn = useMemo(() => {
    if (!debate || !user) {
      return false
    }

    const lastTurn = debate.turn?.at(-1)

    if (!debate.turn || !debate.turn.length || !lastTurn) {
      return debate.creatorSub === user.sub
    }

    return (lastTurn.userSub !== user.sub)
  }, [debate, user])

  const renderNewTurn = useMemo(() => {
    if (!debate) { return }
    if (isCooldownActive) { return 'wait for cooldown' }
    if (!isItYourTurn) { return 'wait for your opponent' }

    return <NewTurn debate={debate} onSubmit={updateDebate} />
  }, [debate, isCooldownActive, isItYourTurn, updateDebate])

  /*useEffect(() => {
    if (isLoading) {
      return
    }
    if (!!user && !!user.sub) {
      updateDebate()
    } else {
      // return <signin/register flow>
      //console.log('no user')
    }
  }, [isLoading, updateDebate, user])*/

  if (isLoading) { return 'loading....' }
  if (error) {
    console.error(error)
    return 'error'
  }
  if (!user) { return 'user not found' }
  if (!debate) { return 'debate not found' }
  /*if (isLoading) { return 'loading user...' }
  if (error) { return JSON.stringify(error) }
  if (!user) { return 'please login' }
  if (!debate) { return 'Debate not found' }*/
  /*if (!debate.opponentSub && (debate.creatorSub === user.sub)) {
    return <Invite debateId={debate.id} />
  }
  if (!debate.opponentSub) {
    return <JoinDebate debateId={debate.id} />
  }*/

  if (debate.opponentSub === null) {
    return (debate.creatorSub === user.sub) ?
    <Invite debateId={debate.id} /> :
    <JoinDebate debateId={debate.id} />
  }

  return (
    <section data-testid="debate-section">
      <h1>{debate.topic}</h1>
      <Turns debate={debate} />
      { renderNewTurn }
    </section>
  )
}
