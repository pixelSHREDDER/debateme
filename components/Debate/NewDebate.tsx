'use client'

import { addDebate } from '@/actions/add-debate';
import { useFormState } from 'react-dom'

const initialState = {
  message: '',
}

export default function NewDebate({ creatorId }: { creatorId: number }) {
  const [state, formAction] = useFormState(addDebate, initialState)

  return (
    <form action={formAction}>
      <input
        type="hidden"
        id="creatorId"
        name="creatorId"
        value={creatorId} />
      <label htmlFor="topic">Topic</label>
      <input type="text" id="topic" name="topic" required />
      <p aria-live="polite" className="sr-only">
        {state?.message}
      </p>
      <button type="submit">Start Debating!</button>
    </form>
  )
}