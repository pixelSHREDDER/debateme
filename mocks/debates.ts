import { TDebate } from '@/lib/prisma-types'
import { DebateStatus } from '@prisma/client'
import { createTurns } from './turns'

export const createDebates: () => TDebate[] = () => {
  const createdAtDate = new Date('2024-04-13T12:24:02Z')
  const cooldownDate = new Date()
  const createdTurns = createTurns()

  return [
    {
      id: 1,
      topic: 'Your turn',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      turn: createdTurns,
      status: DebateStatus.CreatorTurn,
    },
    {
      id: 2,
      topic: 'Opponent\'s Turn',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      status: DebateStatus.OpponentTurn,
      turn: [
        createdTurns[0],
      ],
    },
    {
      id: 3,
      topic: 'Wait for Cooldown',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: 'auth0|abcdefghij1234567890abcd',
      cooldownMins: 60,
      status: DebateStatus.CreatorCooldown,
      turn: [
        createdTurns[0],
        {
          ...createdTurns[1],
          createdAt: cooldownDate,
        },
      ],
    },
    {
      id: 4,
      topic: 'No opponent',
      createdAt: createdAtDate,
      creatorSub: 'auth0|1234567890abcdefghij1234',
      opponentSub: null,
      cooldownMins: 60,
      status: DebateStatus.NoOpponent,
    },
    {
      id: 5,
      topic: 'You\'re Invited to join',
      createdAt: createdAtDate,
      creatorSub: 'auth0|abcdefghij1234567890abcd',
      opponentSub: null,
      cooldownMins: 60,
      status: DebateStatus.NoOpponent,
    },
  ] as const
}
