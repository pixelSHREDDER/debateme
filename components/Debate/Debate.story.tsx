import type { Meta, StoryObj } from '@storybook/react'
import { http, HttpResponse, delay } from 'msw'
import { UserProfile } from '@auth0/nextjs-auth0/client'
import { createDebates } from '@/mocks/debates'
import { getUsers } from '@/mocks/users'
import Debate from './Debate'

const meta: Meta<typeof Debate> = {
  title: 'Debate',
  component: Debate,
}

export default meta

type Story = StoryObj<typeof Debate>

const UserData: UserProfile = getUsers()[0]
const debates = createDebates()
const updateDebate = () => {}

const loginHandler = http.get('http://localhost:6006/api/auth/me', () => HttpResponse.json(UserData))

export const YourTurn: Story = {
  args: {
    debate: debates[0],
    updateDebate,
  },
  parameters: {
    msw: {
      handlers: [
        loginHandler,
      ],
    },
  },
}

export const OpponentsTurn: Story = {
  args: {
    debate: debates[1],
    updateDebate,
  },
  parameters: YourTurn.parameters,
}

export const WaitForCooldown: Story = {
  args: {
    debate: debates[2],
    updateDebate,
  },
  parameters: YourTurn.parameters,
}

export const NoOpponent: Story = {
  args: {
    debate: debates[3],
    updateDebate,
  },
  parameters: YourTurn.parameters,
}

export const InvitedToJoin: Story = {
  args: {
    debate: debates[4],
    updateDebate,
  },
  parameters: YourTurn.parameters,
}

export const Loading: Story = {
  args: YourTurn.args,
  parameters: {
    msw: {
      handlers: [
        http.get(
          'http://localhost:6006/api/auth/me',
          async () => { await delay('infinite') }),
      ],
    },
  },
}

export const Error: Story = {
  args: YourTurn.args,
  parameters: {
    msw: {
      handlers: [
        http.get(
          'http://localhost:6006/api/auth/me',
          () => new HttpResponse(null, { status: 403 })
        ),
      ],
    },
  },
}

export const NotFound: Story = {
  args: {
    debate: null,
    updateDebate,
  },
  parameters: YourTurn.parameters,
}

export const LoggedOut: Story = {
  args: YourTurn.args,
  parameters: {
    msw: {
      handlers: [
        http.get('http://localhost:6006/api/auth/me', async () => {
          await delay(800)
          return new HttpResponse(null, {
            status: 403,
          })
        }),
      ],
    },
  },
}
