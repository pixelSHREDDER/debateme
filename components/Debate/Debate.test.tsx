import * as auth0Hooks from '@auth0/nextjs-auth0/client'
import { afterAll, beforeAll, beforeEach, describe, expect, test, vi } from 'vitest'
import { render, screen } from '@testing-library/react'
import ProviderWrapper, { InviteIdProviderWrapper } from '@/tests/helpers/ProviderWrapper'
import { createDebates } from '@/mocks/debates'
import { getUsers } from '@/mocks/users'
import Debate from './Debate'

const debates = createDebates()
const users: auth0Hooks.UserProfile[] = getUsers()

describe('Debate component', () => {
  const useUserSpy = vi.spyOn(auth0Hooks, 'useUser')

  beforeAll(() => {
    const date = new Date('2024-04-13T12:25:02Z')
    vi.setSystemTime(date)
  })

  beforeEach(() => {
    useUserSpy.mockReturnValue({
      checkSession: vi.fn(),
      isLoading: false,
      user: users[0],
    })
  })

  afterAll(() => {
    vi.useRealTimers()
  })

  test('renders correctly', () => {
    render(
      <Debate debate={debates[0]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByTestId('debate-section')).toMatchSnapshot()
  })

  test('renders invite link screen correctly', () => {
    render(
      <Debate debate={debates[3]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByTestId('invite-section')).toMatchSnapshot()
  })

  test('shows alert when user attempts to join debate without using invite link', () => {
    render(
      <Debate debate={debates[4]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByTestId('debate-no-invite-link-alert')).not.toBeNull()
  })

  test('hides New Turn and renders cooldown message correctly', () => {
    render(
      <Debate debate={debates[2]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.getByTestId('debate-section')).toMatchSnapshot()
  })

  test('does not render debate if user is loading', () => {
    useUserSpy.mockReturnValueOnce({
      checkSession: vi.fn(),
      isLoading: true,
      user: users[0],
    })
    render(
      <Debate debate={debates[0]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.queryByTestId).toBeNull
  })

  test('does not render debate if user has an error', () => {
    useUserSpy.mockReturnValueOnce({
      checkSession: vi.fn(),
      error: new Error(),
      isLoading: false,
      user: users[0],
    })
    render(
      <Debate debate={debates[0]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.queryByTestId).toBeNull
  })

  test('does not render debate if user is not defined', () => {
    useUserSpy.mockReturnValueOnce({
      checkSession: vi.fn(),
      isLoading: false,
    })
    render(
      <Debate debate={debates[0]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.queryByTestId).toBeNull
  })

  test('does not render debate if debate is not defined', () => {
    render(
      <Debate debate={undefined as any} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.queryByTestId).toBeNull
  })

  test('does not render debate if invite id is not in the URL', () => {
    render(
      <Debate debate={debates[0]} updateDebate={vi.fn()} />,
      { wrapper: ProviderWrapper },
    )
    expect(screen.queryByTestId).toBeNull
  })
})
