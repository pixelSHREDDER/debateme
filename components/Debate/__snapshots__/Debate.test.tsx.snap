// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Debate component > hides New Turn and renders cooldown message correctly 1`] = `
<section
  data-testid="debate-section"
>
  <h1>
    Wait for Cooldown
  </h1>
  <ul
    class="_list_7264fa"
  >
    <li
      class="_card_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
      style="--paper-radius: var(--mantine-radius-lg);"
    >
      <div>
        
          
        <p>
          Didn't melt fairer keepsakes since Fellowship elsewhere.
        </p>
        
          
        <p>
          Woodlands payment 
          <a
            href="#"
          >
            Osgiliath
          </a>
           tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
        </p>
        
          
        <p>
          Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain? 
          <span
            style="font-size: 2.0rem"
          >
            Tad survive ensnare joy mistake courtesy Bagshot Row.
          </span>
           Ligulas step drops both? You shall not pass!
        </p>
        
          
        <p>
          Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil. 
          <span
            style="font-size: 0.6rem"
          >
            North valor overflowing sort Iáve mister kingly money?
          </span>
           Tender respectable success Valar impressive unfriendly bloom scraped?
        </p>
        
          
        <ul>
          
            
          <li>
            Adamant.
          </li>
          
            
          <li>
            Southfarthing!
          </li>
          
            
          <li>
            Witch-king.
          </li>
          
            
          <li>
            Precious.
          </li>
          
            
          <li>
            Gaffer's!
          </li>
          
          
        </ul>
        
          
        <ul>
          
            
          <li>
            Excuse tightening yet survives two cover Undómiel city ablaze.
          </li>
          
            
          <li>
            Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
          </li>
          
            
          <li>
            Women rippling cold steps rules Thengel finer.
          </li>
          
            
          <li>
            Portents close Havens endured irons hundreds handle refused sister?
          </li>
          
            
          <li>
            Harbor Grubbs fellas riddles afar!
          </li>
          
          
        </ul>
        
          
        <p>
          Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
        </p>
        
          
        <ol>
          
            
          <li>
            Narsil.
          </li>
          
            
          <li>
            Elros.
          </li>
          
            
          <li>
            Arwen Evenstar.
          </li>
          
            
          <li>
            Maggot's?
          </li>
          
            
          <li>
            Bagginses?
          </li>
          
          
        </ol>
        
          
        <ol>
          
            
          <li>
            Concerning Hobbits l golf air fifth bell prolonging camp.
          </li>
          
            
          <li>
            Grond humble rods nearest mangler.
          </li>
          
            
          <li>
            Enormity Lórien merry gravy stayed move.
          </li>
          
            
          <li>
            Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
          </li>
          
            
          <li>
            Agree ruling um wasteland Bagshot Row expect sleep.
          </li>
          
          
        </ol>
        
          
        <p>
          Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
        </p>
        
          
        <hr />
        
          
        <p>
          What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
        </p>
        
          
        <blockquote>
          
            
          <p>
            You shall be the Fellowship of the Ring.
          </p>
          
            
          <footer>
            —Númenor, 
            <cite>
              sweeter burned verse
            </cite>
          </footer>
          
          
        </blockquote>
        
          
        <p>
          I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
        </p>
        
          
        <pre>
          Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
        </pre>
        
          
        <hr />
      </div>
    </li>
    <li
      class="_card_f5603a _opponentCard_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
      style="--paper-radius: var(--mantine-radius-lg);"
    >
      <div>
        
          
        <p>
          Didn't melt fairer keepsakes since Fellowship elsewhere.
        </p>
        
          
        <p>
          Woodlands payment 
          <a
            href="#"
          >
            Osgiliath
          </a>
           tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
        </p>
        
          
        <p>
          <span
            style="font-size: 2.0rem"
          >
            Tad survive ensnare joy mistake courtesy Bagshot Row.
          </span>
        </p>
        
          
        <p>
          Ligulas step drops both? You shall not pass! Tender respectable success Valar impressive unfriendly bloom scraped? Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil.
        </p>
        
          
        <p>
          <span
            style="font-size: 0.6rem"
          >
            North valor overflowing sort Iáve mister kingly money?
          </span>
        </p>
        
          
        <p>
          Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain?
        </p>
        
          
        <ul>
          
            
          <li>
            Adamant.
          </li>
          
            
          <li>
            Southfarthing!
          </li>
          
            
          <li>
            Witch-king.
          </li>
          
            
          <li>
            Precious.
          </li>
          
            
          <li>
            Gaffer's!
          </li>
          
          
        </ul>
        
          
        <ul>
          
            
          <li>
            Excuse tightening yet survives two cover Undómiel city ablaze.
          </li>
          
            
          <li>
            Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
          </li>
          
            
          <li>
            Women rippling cold steps rules Thengel finer.
          </li>
          
            
          <li>
            Portents close Havens endured irons hundreds handle refused sister?
          </li>
          
            
          <li>
            Harbor Grubbs fellas riddles afar!
          </li>
          
          
        </ul>
        
          
        <p>
          Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
        </p>
        
          
        <ol>
          
            
          <li>
            Narsil.
          </li>
          
            
          <li>
            Elros.
          </li>
          
            
          <li>
            Arwen Evenstar.
          </li>
          
            
          <li>
            Maggot's?
          </li>
          
            
          <li>
            Bagginses?
          </li>
          
          
        </ol>
        
          
        <ol>
          
            
          <li>
            Concerning Hobbits l golf air fifth bell prolonging camp.
          </li>
          
            
          <li>
            Grond humble rods nearest mangler.
          </li>
          
            
          <li>
            Enormity Lórien merry gravy stayed move.
          </li>
          
            
          <li>
            Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
          </li>
          
            
          <li>
            Agree ruling um wasteland Bagshot Row expect sleep.
          </li>
          
          
        </ol>
        
          
        <p>
          Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
        </p>
        
          
        <hr />
        
          
        <p>
          What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
        </p>
        
          
        <blockquote>
          
            
          <p>
            You shall be the Fellowship of the Ring.
          </p>
          
            
          <footer>
            —Númenor, 
            <cite>
              sweeter burned verse
            </cite>
          </footer>
          
          
        </blockquote>
        
          
        <p>
          I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
        </p>
        
          
        <pre>
          Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
        </pre>
      </div>
    </li>
  </ul>
  <div
    aria-describedby="alert_debate_cooldown-body"
    class="m_66836ed3 mantine-Alert-root"
    data-variant="outline"
    id="alert_debate_cooldown"
    role="alert"
    style="--alert-radius: var(--mantine-radius-lg); --alert-bg: transparent; --alert-color: #bfc4d9; --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid #bfc4d9; margin-block: calc(1.25rem * var(--mantine-scale)); background: rgb(240, 243, 253);"
  >
    <div
      class="m_a5d60502 mantine-Alert-wrapper"
    >
      <div
        class="m_667c2793 mantine-Alert-body"
      >
        <div
          class="m_7fa78076 mantine-Alert-message"
          data-variant="outline"
          id="alert_debate_cooldown-body"
        >
          The cooldown period is active. 
          You can start your next turn at 5:25:01 AM. In the meantime, why not read over the previous turns? You never know what you might've missed!
        </div>
      </div>
    </div>
  </div>
</section>
`;

exports[`Debate component > renders correctly 1`] = `
<section
  data-testid="debate-section"
>
  <h1>
    Your turn
  </h1>
  <ul
    class="_list_7264fa"
  >
    <li
      class="_card_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
      style="--paper-radius: var(--mantine-radius-lg);"
    >
      <div>
        
          
        <p>
          Didn't melt fairer keepsakes since Fellowship elsewhere.
        </p>
        
          
        <p>
          Woodlands payment 
          <a
            href="#"
          >
            Osgiliath
          </a>
           tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
        </p>
        
          
        <p>
          Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain? 
          <span
            style="font-size: 2.0rem"
          >
            Tad survive ensnare joy mistake courtesy Bagshot Row.
          </span>
           Ligulas step drops both? You shall not pass!
        </p>
        
          
        <p>
          Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil. 
          <span
            style="font-size: 0.6rem"
          >
            North valor overflowing sort Iáve mister kingly money?
          </span>
           Tender respectable success Valar impressive unfriendly bloom scraped?
        </p>
        
          
        <ul>
          
            
          <li>
            Adamant.
          </li>
          
            
          <li>
            Southfarthing!
          </li>
          
            
          <li>
            Witch-king.
          </li>
          
            
          <li>
            Precious.
          </li>
          
            
          <li>
            Gaffer's!
          </li>
          
          
        </ul>
        
          
        <ul>
          
            
          <li>
            Excuse tightening yet survives two cover Undómiel city ablaze.
          </li>
          
            
          <li>
            Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
          </li>
          
            
          <li>
            Women rippling cold steps rules Thengel finer.
          </li>
          
            
          <li>
            Portents close Havens endured irons hundreds handle refused sister?
          </li>
          
            
          <li>
            Harbor Grubbs fellas riddles afar!
          </li>
          
          
        </ul>
        
          
        <p>
          Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
        </p>
        
          
        <ol>
          
            
          <li>
            Narsil.
          </li>
          
            
          <li>
            Elros.
          </li>
          
            
          <li>
            Arwen Evenstar.
          </li>
          
            
          <li>
            Maggot's?
          </li>
          
            
          <li>
            Bagginses?
          </li>
          
          
        </ol>
        
          
        <ol>
          
            
          <li>
            Concerning Hobbits l golf air fifth bell prolonging camp.
          </li>
          
            
          <li>
            Grond humble rods nearest mangler.
          </li>
          
            
          <li>
            Enormity Lórien merry gravy stayed move.
          </li>
          
            
          <li>
            Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
          </li>
          
            
          <li>
            Agree ruling um wasteland Bagshot Row expect sleep.
          </li>
          
          
        </ol>
        
          
        <p>
          Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
        </p>
        
          
        <hr />
        
          
        <p>
          What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
        </p>
        
          
        <blockquote>
          
            
          <p>
            You shall be the Fellowship of the Ring.
          </p>
          
            
          <footer>
            —Númenor, 
            <cite>
              sweeter burned verse
            </cite>
          </footer>
          
          
        </blockquote>
        
          
        <p>
          I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
        </p>
        
          
        <pre>
          Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
        </pre>
        
          
        <hr />
      </div>
    </li>
    <li
      class="_card_f5603a _opponentCard_f5603a m_e615b15f mantine-Card-root m_1b7284a3 mantine-Paper-root"
      style="--paper-radius: var(--mantine-radius-lg);"
    >
      <div>
        
          
        <p>
          Didn't melt fairer keepsakes since Fellowship elsewhere.
        </p>
        
          
        <p>
          Woodlands payment 
          <a
            href="#"
          >
            Osgiliath
          </a>
           tightening. Barad-dur follow belly comforts tender tough bell? Many that live deserve death. Some that die deserve life. Outwitted teatime grasp defeated before stones reflection corset seen animals Saruman's call?
        </p>
        
          
        <p>
          <span
            style="font-size: 2.0rem"
          >
            Tad survive ensnare joy mistake courtesy Bagshot Row.
          </span>
        </p>
        
          
        <p>
          Ligulas step drops both? You shall not pass! Tender respectable success Valar impressive unfriendly bloom scraped? Branch hey-diddle-diddle pony trouble'll sleeping during jump Narsil.
        </p>
        
          
        <p>
          <span
            style="font-size: 0.6rem"
          >
            North valor overflowing sort Iáve mister kingly money?
          </span>
        </p>
        
          
        <p>
          Curse you and all the halflings! Deserted anytime Lake-town burned caves balls. Smoked lthilien forbids Thrain?
        </p>
        
          
        <ul>
          
            
          <li>
            Adamant.
          </li>
          
            
          <li>
            Southfarthing!
          </li>
          
            
          <li>
            Witch-king.
          </li>
          
            
          <li>
            Precious.
          </li>
          
            
          <li>
            Gaffer's!
          </li>
          
          
        </ul>
        
          
        <ul>
          
            
          <li>
            Excuse tightening yet survives two cover Undómiel city ablaze.
          </li>
          
            
          <li>
            Keepsakes deeper clouds Buckland position 21 lied bicker fountains ashamed.
          </li>
          
            
          <li>
            Women rippling cold steps rules Thengel finer.
          </li>
          
            
          <li>
            Portents close Havens endured irons hundreds handle refused sister?
          </li>
          
            
          <li>
            Harbor Grubbs fellas riddles afar!
          </li>
          
          
        </ul>
        
          
        <p>
          Ravens wonder wanted runs me crawl gaining lots faster! Khazad-dum surprise baby season ranks. I bid you all a very fond farewell.
        </p>
        
          
        <ol>
          
            
          <li>
            Narsil.
          </li>
          
            
          <li>
            Elros.
          </li>
          
            
          <li>
            Arwen Evenstar.
          </li>
          
            
          <li>
            Maggot's?
          </li>
          
            
          <li>
            Bagginses?
          </li>
          
          
        </ol>
        
          
        <ol>
          
            
          <li>
            Concerning Hobbits l golf air fifth bell prolonging camp.
          </li>
          
            
          <li>
            Grond humble rods nearest mangler.
          </li>
          
            
          <li>
            Enormity Lórien merry gravy stayed move.
          </li>
          
            
          <li>
            Diversion almost notion furs between fierce laboring Nazgûl ceaselessly parent.
          </li>
          
            
          <li>
            Agree ruling um wasteland Bagshot Row expect sleep.
          </li>
          
          
        </ol>
        
          
        <p>
          Delay freezes Gollum. Let the Ring-bearer decide. Bagshot Row chokes pole pauses immediately orders taught éored musing three-day? Disease rune repel source fire Goblinses already?
        </p>
        
          
        <hr />
        
          
        <p>
          What about second breakfast? Nags runt near Lindir lock discover level? Andûril breathe waited flatten union.
        </p>
        
          
        <blockquote>
          
            
          <p>
            You shall be the Fellowship of the Ring.
          </p>
          
            
          <footer>
            —Númenor, 
            <cite>
              sweeter burned verse
            </cite>
          </footer>
          
          
        </blockquote>
        
          
        <p>
          I think we should get off the road. Penalty sight splintered Misty Mountain mithril? Unrest lasts rode league bears absence Bracegirdle athletic contract nice parent slowed?
        </p>
        
          
        <pre>
          Pardon Concerning Hobbits rune goblins? Twitching figure including rightful Thorin's level! Worth tubers threats Hornburg deadliest? Unfold thumping shh wants Homely!
        </pre>
      </div>
    </li>
  </ul>
  <form
    data-testid="new-turn-form"
  >
    <fieldset
      class="m_ef274e49 m_e9408a47 mantine-Fieldset-root"
      data-variant="filled"
      radius="lg"
      style="--fieldset-radius: var(--mantine-radius-lg); margin-inline: 0rem;"
    >
      <legend
        class="m_90794832 mantine-Fieldset-legend"
      >
        It's your turn!
      </legend>
      <input
        id="debateId"
        name="debateId"
        type="hidden"
        value="1"
      />
      <input
        id="isCreator"
        name="isCreator"
        type="hidden"
        value="1"
      />
      <input
        id="userSub"
        name="userSub"
        type="hidden"
        value="auth0|1234567890abcdefghij1234"
      />
      <input
        id="bodyString"
        name="bodyString"
        type="hidden"
      />
      <div
        class="m_8bffd616 mantine-Flex-root __m__-r3"
        style="gap: calc(1.25rem * var(--mantine-scale)); align-items: center; justify-content: flex-start;"
      >
        <aside
          aria-describedby="alert_quotes_:r5:-body"
          class="_quotes_e6219c m_66836ed3 mantine-Alert-root"
          data-variant="outline"
          id="alert_quotes_:r5:"
          role="alert"
          style="--alert-radius: var(--mantine-radius-md); --alert-bg: transparent; --alert-color: #9562d2; --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid #9562d2; font-style: italic; width: auto; max-width: 75%;"
        >
          <div
            class="m_a5d60502 mantine-Alert-wrapper"
          >
            <div
              class="m_667c2793 mantine-Alert-body"
            >
              <div
                class="m_7fa78076 mantine-Alert-message"
                data-variant="outline"
                id="alert_quotes_:r5:-body"
              >
                “Don’t be a dick.” ― Unknown
              </div>
            </div>
          </div>
        </aside>
        <button
          class="mantine-focus-auto mantine-active m_77c9d27d mantine-Button-root m_87cf2631 mantine-UnstyledButton-root"
          data-variant="filled"
          style="--button-radius: var(--mantine-radius-md); --button-bg: var(--mantine-color-marble-filled); --button-hover: var(--mantine-color-marble-filled-hover); --button-color: var(--mantine-color-white); --button-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; flex-shrink: 0;"
          type="submit"
        >
          <span
            class="m_80f1301b mantine-Button-inner"
          >
            <span
              class="m_811560b9 mantine-Button-label"
            >
              Finish Your Turn
            </span>
          </span>
        </button>
      </div>
    </fieldset>
  </form>
</section>
`;

exports[`Debate component > renders invite link screen correctly 1`] = `
<section
  data-testid="invite-section"
>
  <h2>
    Invite Your Opponent to Join this Debate
  </h2>
  <div
    class="m_18320242 mantine-Skeleton-root"
    data-animate="true"
  >
    <p>
      Share the link below with your opponent, so they can join and start the debate!
    </p>
  </div>
  <div
    class="m_18320242 mantine-Skeleton-root"
    data-animate="true"
    style="--skeleton-height: calc(3.4375rem * var(--mantine-scale));"
  >
    <div
      aria-describedby="alert_invite_link-body"
      class="m_66836ed3 mantine-Alert-root"
      data-variant="light"
      id="alert_invite_link"
      role="alert"
      style="--alert-radius: var(--mantine-radius-lg); --alert-bg: rgba(125, 62, 201, 0.1); --alert-color: #7d3ec9; --alert-bd: calc(0.0625rem * var(--mantine-scale)) solid transparent; margin-block: calc(1.25rem * var(--mantine-scale));"
    >
      <div
        class="m_a5d60502 mantine-Alert-wrapper"
      >
        <div
          class="m_667c2793 mantine-Alert-body"
        >
          <div
            class="m_7fa78076 mantine-Alert-message"
            data-variant="light"
            id="alert_invite_link-body"
          >
            Remember, this invite link will expire after being used once.
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="m_18320242 mantine-Skeleton-root"
    data-animate="true"
    data-visible="true"
    style="--skeleton-height: calc(2.25rem * var(--mantine-scale)); --skeleton-radius: var(--mantine-radius-sm);"
  />
</section>
`;
